### PR TITLE
Proposal: standardise protocols : only ip:port

### PR DESCRIPTION
**Pull request by vieux**
Let's standardise the way addrs are handles by swarm:

```
--------------                   ---------------------                   ----------------
| swarm join | ----------------> | discovery service | ----------------> | swarm manage |
--------------                   ---------------------                   ----------------
                                           |                             --------------
                                           ----------------------------> | swarm list |
                                                                         --------------
```

#### Swarm join

`swarm join` will only support this format:
* `<ip:port>`

Any other format (`http://`, `https://`, `tcp://`, `foobar://`, etc...) will be denied

#### Swarm manage

`swarm manage` will try to connect using `https` if the `--tls` otherwise `http`

#### Swarm list

`swarm list` will always display addresses as `<ip:port>`

Replace #184
Close #184

### CODE REVIEW

**Comment by SvenDowideit** [comment]
I have some non-encrypted Docker daemons running for testing, and letting me see whats happening, and then many tls ones that I talk to - please don't make it hard to mix them!

**Comment by aluzzardi** [comment]
LGTM

File: discovery/README.md [hunk 1]
```diff
@@ -52,7 +54,9 @@ $ docker -H <swarm_ip:swarm_port> logs ...
 
 # list nodes in your cluster
 $ swarm list --discovery file:///tmp/my_cluster
-http://<node_ip:2375>
+tcp://<node_ip1:2375>
+http://<node_ip2:2375>
+https://<node_ip3:2375>
```
Comment (thaJeztah) [inline discovery/README.md]: Should this be `2376` for secure connections?

**Comment by vieux** [comment]
Ok I updated the issue description and the code.

The cluster needs to be full `http` or full `https` we don't allow mixing the two.

The code is now much simpler.

@aluzzardi please merge when you have time

**Pull request merged**